pkgs/tools/system/hw-probe/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+
+# Required
+, hwinfo
+, dmidecode
+, smartmontools
+, pciutils
+, usbutils
+, edid-decode
+
+# Recommended
+, withRecommended ? false # Install recommended tools
+, mcelog
+, hdparm
+, systemd
+, acpica-tools
+, drm_info
+, mesa-demos
+, memtester
+, sysstat
+, cpuid
+, util-linuxMinimal
+, xinput
+, libva-utils
+, inxi
+, vulkan-utils
+, i2c-tools
+, opensc
+
+# Suggested
+, withSuggested ? false # Install (most) suggested tools
+, hplip
+, sane-backends
+, pnputils ? null # pnputils (lspnp) isn't currently in nixpkgs and appears to be poorly maintained
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hw-probe";
+  version = "1.6.4";
+
+  src = fetchFromGitHub {
+    owner = "linuxhw";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:028wnhrbn10lfxwmcpzdbz67ygldimv7z1k1bm64ggclykvg5aim";
+  };
+
+  makeFlags = [ "prefix=$(out)" ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  makeWrapperArgs =
+    let
+      requiredPrograms = [
+        hwinfo
+        dmidecode
+        smartmontools
+        pciutils
+        usbutils
+        edid-decode
+      ];
+      recommendedPrograms = [
+        mcelog
+        hdparm
+        systemd # (systemd-analyze)
```

Comment:
I don't think it's a good idea to require installation of systemd.

The tool should probe for systemd-analyze only if systemd is already installed and used by the computer owner.

Reply:
Thanks @linuxhw - seeing you're willing drop by motivates me to try and get the PR accepted.

> I don't think it's a good idea to require installation of systemd.
>
> The tool should probe for systemd-analyze only if systemd is already installed and used by the computer owner.

Fair enough, I should probably add a `systemdSupport` input argument to support non-NixOS use cases.

(Note that nix has a slightly different interpretation of "installation" which means something more like "make binaries available in a specific context", rather what I would call "activation". Never-the-less your point is well taken.)

---
Quick suggestion that may require a separate PR to make the change treewide:

How do others feel about adding a flag for `systemdSupport ? config.systemdSupport or stdenv.isLinux`? I.e. `config.systemdSupport` would be similar to [other global "use flags"](https://github.com/NixOS/nixpkgs/issues/116141#issuecomment-798783043) like `config.cudaSupport`.

I've noticed that some other packages have input flags like `systemdSupport ? stdenv.isLinux` or `systemdSupport ? stdenv.hostPlatform.isLinux`. However I've personally worked with nix + non-systemd linux, so I find that default to be a little off target.

Reply:
Related: There are well defined config options now #166792